src/recipe.jl

```diff
@@ -110,6 +113,51 @@ function Makie.plot!(plot::Beeswarm)
         calculate!(point_buffer.val, algorithm, direction == :y ? pixelspace_point_buffer.val : reverse.(pixelspace_point_buffer.val), markersize, side)
         # Project the beeswarm back to data space and store it, again, in `point_buffer.val`
         point_buffer.val .= Point2f.(Makie.project.((scene.camera,), :pixel, :data, direction == :y ? (point_buffer.val) : reverse.(point_buffer.val)))
+
+        # Method to create a gutter when a gutter is defined
+        # NOTE: Maybe turn this into a helper function?
+
+        if !isnothing(gutter)
+            # This gets the x coordinate of all points
+            xs = first.(positions)
+
+            # Find all points belonging to all unique categories
+            # by finding the unique x values
+            idx = 1
+            for group in unique(xs)
+                
+                # Starting index for the group
+                group_indices = findall(==(group), xs)
+
+                # Calculate a gutter threshold
+                gutter_threshold_count = length(group_indices) * gutter_threshold
+                gutter_pts = 0
+                for pt in view(point_buffer.val, group_indices)
```

Comment:
```suggestion
                for idx in group_indices
                    x = xs[idx]
                    pt = point_buffer.val[pt]
```
We can manually deconstruct to avoid having to do `if direction==...` again.